packages/client/hmi-client/src/components/workflow/ops/calibrate-ciemss/tera-calibrate-ciemss-drilldown.vue

```diff
@@ -1145,10 +1145,15 @@ const initialize = async () => {
 	modelPartTypesMap.value = modelPartTypes ?? {};
 
 	// dataset input
-	const { filename, csv, datasetOptions } = await setupDatasetInput(datasetId.value);
-	currentDatasetFileName.value = filename;
-	csvAsset.value = csv;
-	datasetColumns.value = datasetOptions;
+	if (datasetId.value) {
+		const { filename, datasetOptions } = await setupDatasetInput(datasetId.value);
+		currentDatasetFileName.value = filename;
+		datasetColumns.value = datasetOptions;
+
+		setupCsvAsset(datasetId.value).then((csv) => {
+			csvAsset.value = csv;
+		});
```

Comment:
The idea is that I want to get the `datasetOptions` as quickly as I can and not have them tied into the same function call as getting the `csvAsset` which downloads a potentially large file.

The way that I have written this we are making one additional call to `getDataset` if this is a concern I can refactor that

Reply:
Yes I think we should minimize API calls, we already have a constant stream requests for juptyer checks and workflow synchronizations. Having too many API calls will end up clogging the network since you can only have a handful (6 in chrome) running concurrently against a given domain.

Reply:
I disagree, we are doing so many calls to the backend, getting a simple terarium asset is trivial. Downloading the CSV is another issue, but that's not the case here.

Reply:
@YohannParis okay, next time you field performance questions 😄